src/model/nonce.ts

```diff
@@ -32,3 +37,42 @@ export const create = (nonce: Nonce) => (
     singleStringReply,
     falsyResponseToErrorAsync(new Error("Error saving the key"))
   );
+
+export const getNonceFromSignatureInput = (
+  signatureInput: LollipopSignatureInput
+): E.Either<Error, Nonce> =>
+  pipe(
+    signatureInput.match(nonceRegex),
+    O.fromNullable,
+    // take out only the first group of the match
+    O.chainNullableK(matchArray => matchArray.at(1)),
+    O.fold(
+      () => E.left(new Error("Could not retrieve nonce from signature-input.")),
+      flow(
+        Nonce.decode,
+        E.mapLeft(
+          errors =>
+            new Error(
+              `Error while decoding nonce: [${readableReportSimplified(
+                errors
+              )}]`
+            )
+        )
+      )
+    )
+  );
+
+type GetDeleteNonceT = (
+  redis_client: redis.RedisClientType
+) => (nonce: Nonce) => TE.TaskEither<Error, true>;
+export const getDeleteNonce: GetDeleteNonceT = redis_client => nonce =>
```

Comment:
Maybe we should use the same signature pattern of the model operations.
The create action is build to use the method inside an ReaderTaskEither chain `(nonce: Nonce)(client: RedisClientType) => ...`

Reply:
refactored in https://github.com/pagopa/io-functions-fast-login/pull/27/commits/f31eed4e20119c91d4e2b9977660f7823ed56d1a